src/providers/pkcs11_provider/mod.rs

```diff
@@ -722,9 +727,24 @@ impl Provide for Pkcs11Provider {
                     error!("Error when extracting attribute: {}.", rv);
                     Err(ResponseStatus::PsaErrorCommunicationFailure)
                 } else {
+                    let mut modulus = attrs[0].get_bytes();
+                    let mut public_exponent = attrs[1].get_bytes();
+
+                    // To produce a valid ASN.1 RSAPublicKey structure, 0x00 is put in front of the positive
+                    // integer if highest significant bit is one, to differentiate it from a negative number.
+                    if modulus[0] & 0x80 == 0x80 {
+                        modulus.insert(0, 0x00);
+                    }
+                    if public_exponent[0] & 0x80 == 0x80 {
+                        public_exponent.insert(0, 0x00);
+                    }
+
+                    let modulus = IntegerAsn1::from_signed_bytes_be(modulus);
```

Comment:
I find it weird that they implemented both `as_bytes_be` and `as_signed_bytes_be`, but only `from_signed_bytes_be`. Feels like a method that does the above check and insert before the conversion should exist too... ohwell

Reply:
Asked this in Devolutions/picky-rs#12